base/sparse/sparsematrix.jl

```diff
@@ -27,6 +27,25 @@ end
 
 size(S::SparseMatrixCSC) = (S.m, S.n)
 
+# Define a union of SparseMatrixCSC and view containing all rows of a SparseMatrixCSC and a contiguous
+# selection of columns. Many methods can be defined efficiently for this union by extracting the fields
+# via the get function: getcolptr, getrowval, and getnzval. The key insight is that getcolptr on a view
+# returns a offset view of the colptr of the underlying SparseMatrixCSC
+const SparseMatrixCSCUnion{Tv,Ti} =
+    Union{SparseMatrixCSC{Tv,Ti},
+        SubArray{Tv,2,SparseMatrixCSC{Tv,Ti},Tuple{Base.Slice{Base.OneTo{Int64}},I}}} where {I<:AbstractUnitRange}
+
+getcolptr(S::SparseMatrixCSC) = S.colptr
+getcolptr(S::SubArray{T,2,<:SparseMatrixCSC,Tuple{Base.Slice{Base.OneTo{Int64}},I}} where {T,I<:AbstractUnitRange}) =
+    view(S.parent.colptr, first(indices(S, 2)):(last(indices(S, 2)) + 1))
+getrowval(S::SparseMatrixCSC) = S.rowval
+getrowval(S::SubArray{T,2,<:SparseMatrixCSC,Tuple{Base.Slice{Base.OneTo{Int64}},I}} where {T,I<:AbstractUnitRange}) =
+    S.parent.rowval
+getnzval(S::SparseMatrixCSC) = S.nzval
+getnzval(S::SubArray{T,2,<:SparseMatrixCSC,Tuple{Base.Slice{Base.OneTo{Int64}},I}} where {T,I<:AbstractUnitRange}) =
+    S.parent.nzval
```

Comment:
Lovely! :)

Perhaps worth giving this form of view a name, to avoid repeating the rather complex definition?